client/app/bundles/course/assessment/submission/components/ScribingView/ToolDropdown.jsx

```diff
@@ -40,44 +42,79 @@ const style = {
     fontSize: '12px',
     padding: '10px 0px 10px 0px',
   },
+  disabled: {
+    cursor: 'not-allowed',
+    pointerEvents: 'none',
+    color: '#c0c0c0',
+  },
 };
 
 export default class ToolDropdown extends Component {
   renderIcon() {
-    const { iconClassname, currentTool, toolType, iconComponent } = this.props;
+    const { disabled, iconClassname, currentTool, toolType, iconComponent } = this.props;
+    const iconStyle = disabled ? style.disabled :
+      { color: currentTool === toolType ? blue500 : 'rgba(0, 0, 0, 0.4)' };
 
     return iconComponent ?
       iconComponent() :
       <FontIcon
         className={iconClassname}
-        style={
-          currentTool === toolType ?
-            { color: blue500 } :
-            { color: 'rgba(0, 0, 0, 0.4)' }
-        }
+        style={iconStyle}
       />;
   }
 
   renderColorBar() {
-    const { colorBar, colorBarComponent } = this.props;
+    const { activeObject, disabled, colorBarBorder, colorBarBackground } = this.props;
+
+    let backgroundColor = colorBarBackground;
+    let borderColor = colorBarBorder;
+
+    if (activeObject) {
+      switch (activeObject.type) {
+        case 'path':
+        case 'line':
+          backgroundColor = activeObject.stroke;
+          break;
+        case 'i-text':
+          backgroundColor = activeObject.fill;
+          break;
+        case 'rect':
+        case 'ellipse':
+          backgroundColor = activeObject.fill;
+          borderColor = activeObject.stroke;
+          break;
+        default:
+      }
+    }
+
+    const colorBarStyle = disabled ?
+    {
+      width: '23px',
+      height: '8px',
+      background: '#c0c0c0',
+    } :
+    {
+      width: '23px',
+      height: '8px',
```

Comment:
If you prefer not to use `style.colorBar`, can remove it? It doesn't seem to be used anywhere